projects/wave-core-new/src/lib/operators/dialogs/column-range-filter/column-range-filter.component.ts

```diff
@@ -0,0 +1,252 @@
+import {Component, OnInit, ChangeDetectionStrategy, OnDestroy} from '@angular/core';
+import {ResultTypes} from '../../result-type.model';
+import {FormArray, FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {of, ReplaySubject, Subscription} from 'rxjs';
+import {ProjectService} from '../../../project/project.service';
+import {map, tap, mergeMap} from 'rxjs/operators';
+import {Layer, VectorLayer} from '../../../layers/layer.model';
+import {VectorLayerMetadata} from '../../../layers/layer-metadata.model';
+import {VectorColumnDataTypes} from '../../datatype.model';
+import {WorkflowDict} from '../../../backend/backend.model';
+import {ClusteredPointSymbology, PointSymbology} from '../../../layers/symbology/symbology.model';
+import {colorToDict} from '../../../colors/color';
+import {RandomColorService} from '../../../util/services/random-color.service';
+import {ColumnRangeFilterDict} from '../../../backend/operator.model';
+import {Vector} from 'ol/source';
+
+@Component({
+    selector: 'wave-column-range-filter',
+    templateUrl: './column-range-filter.component.html',
+    styleUrls: ['./column-range-filter.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ColumnRangeFilterComponent implements OnInit, OnDestroy {
+    readonly inputTypes = ResultTypes.VECTOR_TYPES;
+
+    attributes$ = new ReplaySubject<Array<string>>(1);
+
+    form: FormGroup;
+
+    private subscriptions: Array<Subscription> = [];
+    public attributeError: boolean = false;
+    public errorHint: string = 'default error';
+    private columnTypes = new Map<string, string | undefined>();
```

Comment:
can you maybe use the `VectorColumnDataTypes` like `VectorColumnDataTypes.Float` instead of strings here as values of the map?

Reply:
When I do `metadata.dataTypes.get()` for an attribute I'll get `TextColumn`, `FloatColumn` or `IntColumn`. Can I use these three? They seem to fit. Otherwise, how could I get `VectorColumnDataTypes`?

Reply:
I changed it so it's using `VectorColumnDataType` (not type**s**) right now in the Map. But later I still have to extract a string, like this `return this.columnTypes.get(attribute)?.code === 'text';` via the `code` attribute. I couldn't find a way to avoid the string comparison... can you help me with this?

Reply:
you should be able to do the comparisong like `this.columnTypes.get(attribute)?.code === VectorColumnDataTypeCollection.Text`. Then you don't have to compare with a hardcoded string.

Reply:
I see, thanks! All the different classes had me a bit confused, but it works fine now.